changelogutils/changelog.go

```diff
@@ -74,16 +76,74 @@ func GetProposedTag(fs afero.Fs, latestTag, changelogParentPath string) (string,
 	return proposedVersion, nil
 }
 
-func ReadChangelogFile(path string) (*Changelog, error) {
-	var changelog Changelog
-	bytes, err := ioutil.ReadFile(path)
+func ReadChangelogFile(fs afero.Fs, path string) (*ChangelogFile, error) {
+	var changelog ChangelogFile
+	bytes, err := afero.ReadFile(fs, path)
 	if err != nil {
 		return nil, errors.Wrapf(err, "failed reading changelog file: %s", path)
 	}
 
-	if err := yaml.Unmarshal(bytes, changelog); err != nil {
-		return nil, errors.Wrap(err, "failed parsing changelog file")
+	if err := yaml.Unmarshal(bytes, &changelog); err != nil {
+		return nil, errors.Errorf("File %s is not a valid changelog file", path)
 	}
 
 	return &changelog, nil
 }
+
+func VersionToString(v *versionutils.Version) string {
```

Comment:
nit: make this a property of Version instead so it also the fmt stringer interface
func (v *Version) String() string